pages/api/company-overview/[symbol].ts

```diff
@@ -0,0 +1,34 @@
+import axios from "axios"
+import { ICompanyOverview } from "../../../models/CompanyOverview"
+
+// should these be extracted out since they repeat in all of these places?
+const BASE_URL = "https://cloud.iexapis.com"
+const TOKEN = "pk_63509e5b43384ab08845be28759fb5ea"
+
+
+const fetcher = async (symbol: string) => {
+  try {
+    const [r1, r2] = await Promise.all([
```

Comment:
💯 - if we were to make this fancier / more reactive, what you could is, rather than wait for both requests with Promise.all, keep the connection open and send back data as you get it.  The interface would be the same, you'd still send the merge to the UI, but the first payload to the UI would be partial.  After that, you could, either always send the UI deltas/diffs, or keep this layer sending a snapshot to the UI of the "state of the world", caching the part of data that hasn't changed.  None of this is justified at the moment, but just things to think about.

Reply:
Interesting! I might try dabbling with this tho I'm not 100% sure what this actually looks like as code
> After that, you could, either always send the UI deltas/diffs, or keep this layer sending a snapshot to the UI of the "state of the world", caching the part of data that hasn't changed.